### ISSUE TITLE
An emqtt kafka bridge plugin

### ISSUE THREAD

**Issue by vowstar** [emqtt/emqtt_kafka#1]
The official plugin has not been updated for 10 months. So, I have developed a plug-in named emqttd_plugin_kafka_bridge, which can publish messages from emqttd to kafka topic

https://github.com/vowstar/emqttd_plugin_kafka_bridge

Enjoy it!

**Comment by drasko** [comment]
@vowstar if I understand correctly, this plugin takes all the MQTT messages from eMQTT and sends them to Kafka. It also adds some administrative messages (like when client connects/disconnects) from eMQTT's internals and sends these to Kafka also.

Is this correct?

**Comment by vowstar** [comment]
@drasko Yes you're right, emqttd_plugin_kafka_bridge just do these below.

In the following circumstances, you will receive kafka messages
* when a client connected broker
* when a client disconnected broker
* when a client subscribed a channel
* when a client unsubscribed a channel
* when a client published a message to a channel
* when a client delivered a message
* when a client acknowledged a messages

**Comment by drasko** [comment]
@vowstar if I understand correctly, you have implemented this solution so we can scale - i.e. to override problem I mention here: https://groups.google.com/forum/#!topic/rabbitmq-users/KVMNkAsW-ac

Also, why did you need Kafka (what was your use case?) as you could probably leak all the messages from eMQTT to Cassandra or some other database?

**Comment by vowstar** [comment]
@drasko Yes, I think it could do this, but need improved to stable work.

I use Kafka as message queue to solve data storage problem. Large amount of messages will kill the I/O and harddisk, so I need Kafka to buffer these data and let data workers have time to process these data and store it in databases. 

It's possible to forward all the messages to cassandra, I think this [CQErl](https://github.com/matehat/cqerl) could help us to do it, we should replace ``eKaf`` by ``CQErl`` in kafka bridge plugin.
